# Analyze the Political Orientation of Media Outlets in the Mediacloud Dataset

This notebook explores the **political orientation of news domains** using dataset that you can download [here (GitHub)](https://github.com/amirzon10/AI-Regulation-Discourse-in-News-A-Longitudinal-Study-Post-ChatGPT-2020-2025-/blob/main/data/mediacloud-2020-2025-dataset.csv). 

The orientations are categorized as:
- `lean-left`
- `left`
- `center`
- `lean-right`
- `right`

In [1]:
import pandas as pd
import matplotlib.pyplot as plt

### Dataset Overview

We collected thousands of domains and mapped them to their political orientation.  
This helps us understand the distribution of media bias globally across different outlets.

In [2]:
#load your dataset
df = pd.read_csv("data/mediacloud-2020-2025-dataset.csv")
df.head()

id  \
0  1dc171e7750c319dc4a7b4ca87c8a6f5587c0e9481f129...   
1  e6a47f0c7b6de768d799e603b5797ca432f9af6903de7b...   
2  8061d40a29f0fcf33e6906573a72389a4c0a0cc136522c...   
3  d411e1a49c87e2e710c054d514f58beebcfa2f275107d8...   
4  c8ce06691fdf797a875be76c82efb7b0fab0d3d62a795c...   

                       indexed_date language      media_name       media_url  \
0  2025-11-03 00:22:19.404264+00:00       en  techcrunch.com  techcrunch.com   
1  2025-11-02 23:17:22.553191+00:00       en    livemint.com    livemint.com   
2  2025-11-02 21:51:34.827292+00:00       en      apnews.com      apnews.com   
3  2025-11-02 20:26:57.088674+00:00       es    infolibre.es    infolibre.es   
4  2025-11-02 20:17:51.246694+00:00       en     thestar.com     thestar.com   

  publish_date                                              title  \
0   2025-11-02  Google pulls Gemma from AI Studio after Senato...   
1   2025-11-02  Here's why India’s AI content draft rules miss...   
2   2025-11-02  Who is Zico Kolter? A professor leads OpenAI s...   
3   2025-11-02  ChatGPT no es un psicólogo, pero cambiará su l...   
4   2025-11-02  Like maple syrup and hockey, AI must become a ...   

                                                 url  
0  https://techcrunch.com/2025/11/02/google-pulls...  
1  https://www.livemint.com/opinion/online-views/...  
2  https://apnews.com/article/openai-safety-chatg...  
3  https://www.infolibre.es/politica/chatgpt-rect...  
4  https://www.thestar.com/business/opinion/like-...

In [3]:
#check the dataset size
df.shape

#check language distribution
language_size = df['language'].size
print(f"Total entries in language column: {language_size}")
language_counts = df['language'].value_counts()
print("Language distribution:")
print(language_counts)


Total entries in language column: 35399
Language distribution:
en    28928
de     1724
es      831
pl      807
it      631
fr      466
ja      368
zh      316
sv      284
nl      247
ro      209
pt      131
id       84
hr       43
fi       36
hu       36
no       33
tr       30
ru       29
sq       17
ko       16
hi       15
th       14
bg       12
el       11
ca       11
fa        9
tl        8
la        8
uk        7
cs        5
he        4
ar        4
gl        3
is        3
sr        3
mk        2
ur        2
se        2
sw        2
ta        1
sk        1
nn        1
nb        1
ml        1
ka        1
mt        1
az        1
Name: language, dtype: int64


In [4]:
#creating a simple bias mapping (will be expanded)
bias_map = {
    "theguardian.com": "left",
    "nytimes.com": "left",
    "cnn.com": "lean left",
    "bbc.co.uk": "center",
    "reuters.com": "center",
    "foxnews.com": "lean right",
    "wsj.com": "lean right",
    "breitbart.com": "right",
    "abcnews.go.com": "lean left",
    "alternet.org": "left",
    "apnews.com": "center",
    "yahoo.com": "lean left",
    "nbcnews.com": "lean left",
    "thehill.com": "center",
    "theepochtimes.com": "right",
    "seattletimes.com": "center",
    "forbes.com": "center",
    "thetimes.co.uk": "right",
    "msnbc.com": "left",
    "huffpost.com": "left",
    "vox.com": "left",
    "npr.org": "center",
    "usatoday.com": "center",
    "bloomberg.com": "center",
    "washingtonpost.com": "lean left",
    "nationalreview.com": "right",
    "dailycaller.com": "right",
    "newsmax.com": "right",
    "thefederalist.com": "right",
    "propublica.org": "lean left",
    "politico.com": "center",
    "axios.com": "center",
    "newyorker.com": "left",
    "motherjones.com": "left",
    "townhall.com": "right",
    "reason.com": "lean right",
    "slate.com": "left",
    "theatlantic.com": "lean left",
    "pjmedia.com": "right",
    "washingtontimes.com": "right",
    "thebulwark.com": "center",
    "justthenews.com": "right",
    "insider.com": "lean left",
    "cbsnews.com": "center",
    "dailymail.co.uk": "right",
    "marketwatch.com": "center",
    "businessinsider.com": "lean left",
    "buzzfeednews.com": "left",
    "fivethirtyeight.com": "center",
    "realclearpolitics.com": "center",
    "theintercept.com": "left",
    "truthout.org": "left",
    "spectator.org": "right",
    "americanthinker.com": "right",
    "dw.com": "center",
    "rte.ie": "center",
    "straitstimes.com": "center",
    "scmp.com": "center",
    "telegraph.co.uk": "lean right",
    "skynews.com.au": "lean right",
    "spectator.co.uk": "lean right",
    "nationalpost.com": "lean right",
    "jpost.com": "lean right",
    "rt.com": "right",
    "sputniknews.com": "right",
    "israelnationalnews.com": "right",
    "lemonde.fr": "left",
    "elperiodico.com": "left",
    "derfreitag.de": "left",
    "jacobinmag.com": "left",
    "ilmanifesto.it": "left",
    "cbc.ca": "lean left",
    "abc.net.au": "lean left",
    "globalnews.ca": "lean left",
    "theage.com.au": "lean left",
    "irishmirror.ie": "lean left",
    "stuff.co.nz": "lean left",
    "techcrunch.com": "center",
    "livemint.com": "center",
    "infolibre.es": "left",
    "bt.no": "center",
    "jang.com.pk": "lean right",
    "voc.com.cn": "right",
    "thestar.com": "lean left",
    "fortune.com": "center",
    "elplural.com": "left",
    "malaysianews.net": "center",
    "telegraphindia.com": "center",
    "marianne.net": "left",
    "newsday.com": "center",
    "financialexpress.com": "center",
    "globenewswire.com": "center",
    "ts.fi": "center",
    "kuna.net.kw": "right",
    "courrier-picard.fr": "center",
    "independent.co.uk": "lean left",       
    "rri.ro": "center",                     
    "inquirer.com": "lean left",
    "metbuat.az": "center",
    "eldiario.com.ar": "left",
    "nuevodiarioweb.com.ar": "center",
    "gizmodo.jp": "center",
    "malaymail.com": "center",
    "winnipegfreepress.com": "lean left",
    "dubaichronicle.com": "center",
    "necocheanet.com.ar": "center",
    "nigerianpilot.news": "lean right",
    "levante-emv.com": "center",
    "aa.com.tr": "right",
    "elperiodicomediterraneo.com": "center",
    "channelnewsasia.com": "center",
    "lavanguardia.com": "lean left",
    "gbnews.com": "right",
    "expansion.com": "center",
    "tageszeitung.it": "left",
    "pakistantoday.com.pk": "center",
    "indiatimes.com": "center",
    "benzinga.com": "center",
    "app.com.pk": "center",
    "techradar.com": "center",
    "businesstimes.com.sg": "center",
    "krytykapolityczna.pl": "left",
    "today.az": "lean right",
    "codezine.jp": "center",
    "sandiegouniontribune.com": "lean left",
    "livescience.com": "center",
    "vice.com": "left",
    "muvitv.com": "center",
    "cronista.com": "center",
    "wellandtribune.ca": "center",
    "niagarafallsreview.ca": "lean left",
    "thespec.com": "lean left",
    "cbj.ca": "center",
    "therecord.com": "lean left",
    "ocregister.com": "lean right",
    "medianama.com": "lean left",
    "syracuse.com": "center",
    "nyunews.com": "center",
    "okdiario.com": "right",
    "einpresswire.com": "center",
    "itweb.co.za": "center",
    "businessday.ng": "center",
    "iafrica.com": "center",
    "diepresse.com": "center",
    "guardian.ng": "lean left",
    "hindustantimes.com": "center",
    "contrepoints.org": "right",
    "thisdaylive.com": "center",
    "ghanamma.com": "center",
    "newsghana.com.gh": "center",
    "bursa.ro": "center",
    "focus.de": "center",
    "ilsole24ore.com": "center",
    "thebulletin.be": "center",
    "wired.com": "left",
    "cnet.com": "center",
    "heise.de": "center",
    "theglobeandmail.com": "center",
    "merca20.com": "center",
    "arstechnica.com": "center",
    "hollywoodreporter.com": "center",
    "gizmodo.com": "lean left",
    "nation.com.pk": "lean right",
    "thebftonline.com": "center",
    "telegraf.rs": "center",
    "dailycampus.com": "center",
    "dailytimes.com.pk": "center",
    "thenationonlineng.net": "lean right",
    "tasnimnews.com": "right",
    "standard.co.uk": "lean left",
    "hsbnoticias.com": "center",
    "diariodelsur.com.co": "center",
    "engadget.com": "center",
    "manilatimes.net": "lean right",
    "baltictimes.com": "center",
    "sky.com": "center",
    "zdnet.com": "center",
    "rollingstone.com": "lean left",
    "vnanet.vn": "right",
    "nypost.com": "lean right",
    "news18.com": "center",
    "tengrinews.kz": "center",
    "columbiachronicle.com": "center",
    "thehindubusinessline.com": "center",
    "latimes.com": "lean left", 
    "twincities.com": "center",
    "pilotonline.com": "center",
    "theverge.com": "lean left",
    "courant.com": "lean left",
    "mashable.com": "lean left",
    "orlandosentinel.com": "center",
    "sun-sentinel.com": "center",
    "cnbc.com": "center",
    "enca.com": "center",
    "jpnn.com": "center",
    "indianexpress.com": "center",
    "deccanchronicle.com": "center",
    "thevanguardusa.com": "right",
    "thewest.com.au": "center",
    "prnewswire.com": "center",
    "bbc.com": "center",
    "ibtimes.com": "center",
    "234radio.com": "center",
    "legit.ng": "center",
    "rappler.com": "lean left",
    "ibtimes.co.uk": "center",
    "observer.com": "lean left",
    "india.com": "center",
    "yaledailynews.com": "center",
    "businessinsider.jp": "center",
    "prtimes.jp": "center",
    "bicollegenews.com": "center",
    "bhaskar.com": "center",
    "udn.com": "center",
    "faz.net": "center",
    "newsweek.com": "center",
    "theinfostride.com": "center",
    "denverpost.com": "lean left",
    "cityam.com": "lean right",
    "techcabal.com": "center",
    "aljazeera.com": "lean left",
    "nation.africa": "center",
    "stol.it": "center",
    "ibtimes.co.in": "center",
    "boomlive.in": "center",
    "freepressjournal.in": "lean left",
    "bernama.com": "center",
    "dailyfreepress.com": "center",
    "theconversation.com": "center",
    "nzherald.co.nz": "center",
    "thefrontierpost.com": "center",
    "timesofmalta.com": "center",
    "tribunnews.com": "center",
    "adnkronos.com": "center",
    "collegian.com": "center",
    "bangkokpost.com": "center",
    "thestreet.com": "center",
    "ammonnews.net": "right",
    "epochtimes.de": "right",
    "arkansasonline.com": "center",
    "bzi.ro": "center",
    "l-a.no": "center",
    "heraldscotland.com": "lean left",
    "abc.es": "right",
    "libertaddigital.com": "right",
    "malaya.com.ph": "center",
    "monitorulbt.ro": "center",
    "24-7pressrelease.com": "center",
    "breizh-info.com": "right",
    "prnewswire.co.uk": "center",
    "dailythepatriot.com": "lean right",
    "medicinehatnews.com": "center",
    "lethbridgeherald.com": "center",
    "ansa.it": "center",
    "nrc.nl": "center",
    "iltempo.it": "right",
    "alleghenycampus.com": "center",
    "jsonline.com": "lean left",
    "geo.tv": "center",
    "nationalobserver.com": "lean left",
    "ildenaro.it": "center",
    "kessbenonline.com": "center",
    "sbs.com.au": "center",
    "bworldonline.com": "center",
    "thehindu.com": "center",
    "rebelnews.com": "right",
    "nannews.ng": "center",
    "kelownadailycourier.ca": "center",
    "chicagomaroon.com": "center",
    "newsx.com": "center",
    "pentictonherald.ca": "center",
    "vision1fm.com": "center",
    "bostonglobe.com": "lean left",
    "stabroeknews.com": "center",
    "elpais.com": "lean left",
    "postandcourier.com": "lean right",
    "opisantacruz.com.ar": "center",
    "3szek.ro": "center",
    "tnp.sg": "center",
    "investigaction.net": "left",
    "thequint.com": "lean left",
    "thechronicle.com": "center",
    "metro.co.uk": "center",
    "time.com": "lean left",
    "standardmedia.co.ke": "center",
    "chiangraitimes.com": "center",
    "kurier.at": "center",
    "rp.pl": "center",
    "edestad.nl": "center",
    "parismatch.com": "center",
    "philstar.com": "center",
    "temple-news.com": "center",
    "badgerherald.com": "center",
    "dailyrecord.co.uk": "lean left",
    "tribuneonlineng.com": "center",
    "economicconfidential.com": "center",
    "irishexaminer.com": "center",
    "scoop.co.nz": "center",
    "yehey.com": "center",
    "boe.es": "center",
    "jagran.com": "center",
    "zerohedge.com": "right",
    "championnews.com.ng": "lean right",
    "opiniaoenoticia.com.br": "center",
    "hvg.hu": "center",
    "newsdiaryonline.com": "center",
    "urdupoint.com": "center",
    "elperiodicoextremadura.com": "center",
    "politicshome.com": "center",
    "theblaze.com": "right",
    "tvp.info": "right",
    "detik.com": "center",
    "ltn.com.tw": "center",
    "bohuslaningen.se": "center",
    "ziaruldeolt.ro": "center",
    "island.lk": "center",
    "repubblica.it": "lean left",
    "shqip.al": "center",
    "monitor.al": "center",
    "gazzettadelsud.it": "center",
    "timesofisrael.com": "center",
    "gds.ro": "center",
    "noticiasdenavarra.com": "center",
    "arynews.tv": "lean right",
    "ilfoglio.it": "center",
    "nationalnetworkonline.com": "center",
    "loyolaphoenix.com": "center",
    "thedaonline.com": "center",
    "ipanewspack.com": "center",
    "thehoya.com": "center",
    "ghanabusinessnews.com": "center",
    "dr1.com": "center",
    "premiumtimesng.com": "center",
    "dailycollegian.com": "center",
    "theithacan.org": "center",
    "kristianstadsbladet.se": "center",
    "nzz.ch": "center",
    "liwwadders.nl": "center",
    "nsk.se": "center",
    "radio.cz": "center",
    "pakistanpressfoundation.org": "center",
    "therakyatpost.com": "center",
    "dominicantoday.com": "center",
    "maszol.ro": "center",
    "monitorulcj.ro": "center",
    "dziennik.pl": "center",
    "von.gov.ng": "center",
    "dailypost.ng": "center",
    "pressetext.com": "center",
    "bta.bg": "center",
    "newsday.co.tt": "center",
    "blueprint.ng": "center",
    "journal.com.ph": "center",
    "elojodigital.com": "center",
    "theweek.in": "lean left",
    "jungefreiheit.de": "right",
    "icelandreview.com": "center",
    "citizen.co.za": "center",
    "elsaltodiario.com": "left",
    "latribune.fr": "center",
    "globo.com": "center",
    "taipeitimes.com": "center",
    "correiobraziliense.com.br": "center",
    "thedartmouth.com": "center",
    "terra.com.br": "center",
    "curier.ro": "center",
    "oann.com": "right",
    "theinsightnewsonline.com": "center",
    "ziarulargesul.ro": "center",
    "radiodon.com.ar": "center",
    "todaygh.com": "center",
    "baltimoresun.com": "lean left",
    "kyodo.co.jp": "center",
    "dnaindia.com": "center",
    "yle.fi": "center",
    "rollcall.com": "center",
    "pressat.co.uk": "center",
    "vanityfair.com": "lean left",
    "stltoday.com": "center",
    "krone.at": "right",
    "independent.ng": "center",
    "ascii.jp": "center",
    "thegatewaypundit.com": "right",
    "stern.de": "lean left",
    "mitti.se": "center",
    "lopinion.fr": "center",
    "gazetanord-vest.ro": "center",
    "nachrichten.at": "center",
    "cash.ch": "center",
    "sn.at": "center",
    "tagesspiegel.de": "lean left",
    "zeit.de": "lean left",
    "swarajyamag.com": "right",
    "brandonsun.com": "center",
    "crikey.com.au": "left",
    "mustangnews.net": "center",
    "nairametrics.com": "center",
    "upi.com": "center",
    "flathatnews.com": "center",
    "moroccoworldnews.com": "center",
    "dailytribune.com": "center",
    "africannewsagency.com": "center",
    "brisbanetimes.com.au": "lean left",
    "smh.com.au": "lean left",
    "thefinancialdaily.com": "center",
    "centralmaine.com": "center",
    "planbnoticias.com.ar": "center",
    "derstandard.at": "lean left",
    "diariotextual.com": "center",
    "scroll.in": "lean left",
    "dnevnik.rs": "center",
    "lefigaro.fr": "lean right",
    "corriereirpinia.it": "center",
    "newsmirrorng.com": "center",
    "excelsior.com.mx": "center",
    "avvir.no": "center",
    "antiguatribune.com": "center",
    "vikingfusion.com": "center",
    "dailynorthwestern.com": "center",
    "metrotvonline.com": "center",
    "bowdoinorient.com": "center",
    "svnews.ro": "center",
    "dailytrojan.com": "center",
    "industriemagazin.at": "center",
    "20min.ch": "center",
    "capital.ro": "center",
    "dutchnews.nl": "center",
    "dagsavisen.no": "lean left",
    "ekathimerini.com": "center",
    "hotnews.ro": "center",
    "tiempodesanjuan.com": "center",
    "utdailybeacon.com": "center",
    "cursdeguvernare.ro": "center",
    "thewalrus.ca": "left",
    "michigandaily.com": "lean left",
    "coyotechronicle.net": "center",
    "index.hu": "lean left",
    "newsone.com": "lean left",
    "cna.com.tw": "center",
    "irishtimes.com": "lean left",
    "mercurynews.com": "lean left",
    "bfmtv.com": "center",
    "japantoday.com": "center",
    "elcorreo.com": "center",
    "un.org": "center",
    "mynavi.jp": "center",
    "tribune.net.ph": "center",
    "kenyanews.go.ke": "center",
    "diariosigloxxi.com": "center",
    "spotlightnepal.com": "center",
    "ansarpress.com": "lean right",
    "co.za": "center",
    "icirnigeria.org": "center",
    "gazetadecluj.ro": "center",
    "thesandb.com": "center",
    "lavozdegalicia.es": "center",
    "vastranyland.fi": "center",
    "onet.pl": "center",
    "rawstory.com": "left",
    "dn.se": "center",
    "thenews.com.pk": "center",
    "svd.se": "lean right",
    "billboard.com": "center",
    "welt.de": "lean right",
    "di.se": "center",
    "dawn.com": "center",
    "gazetaprawna.pl": "center",
    "schwartzreport.net": "left",
    "theonlineclarion.com": "center",
    "vrt.be": "center",
    "sdpnoticias.com": "center",
    "tvn24.pl": "center",
    "pakobserver.net": "lean right",
    "thelocal.it": "center",
    "srf.ch": "center",
    "kbs.co.kr": "center",
    "nbcsvg.com": "center",
    "brecorder.com": "center",
    "khaama.com": "center",
    "taz.de": "left",
    "wnd.com": "right",
    "mirror.co.uk": "lean left",
    "knack.be": "lean left",
    "timeslive.co.za": "center",
    "jamaica-gleaner.com": "center",
    "laverita.info": "right",
    "mid-day.com": "center",
    "royalgazette.com": "center",
    "affaritaliani.it": "center",
    "investors.com": "lean right",
    "mediaindonesia.com": "center",
    "iwnsvg.com": "center",
    "orf.at": "center",
    "vol.at": "center",
    "thebridge.jp": "center",
    "independent.ie": "center",
    "ilmessaggero.it": "center",
    "thenation.com": "left",
    "lasicilia.it": "center",
    "wfuogb.com": "center",
    "albeu.com": "center",
    "timesindonesia.co.id": "center",
    "panorama.com.al": "center",
    "azcentral.com": "lean left",
    "hongkongfp.com": "center",
    "challenges.fr": "center",
    "katadata.co.id": "center",
    "24.kg": "center",
    "newstalkzb.co.nz": "lean right",
    "elmundo.es": "lean right",
    "cincinnati.com": "center",
    "newsroom.co.nz": "lean left",
    "newcastillian.com": "center",
    "dailyegyptian.com": "center",
    "melillahoy.es": "center",
    "newindianexpress.com": "center",
    "thenews-chronicle.com": "center",
    "elfinanciero.com.mx": "center",
    "leadership.ng": "lean right",
    "cmcforum.com": "center",
    "searchlight.vc": "center",
    "newsweek.pl": "center",
    "case.edu": "center",
    "browndailyherald.com": "lean left",
    "uol.com.br": "center",
    "thesewaneepurple.org": "center",
    "cronicadecantabria.com": "center",
    "theeagleonline.com.ng": "center",
    "telegraaf.nl": "lean right",
    "adevarul.ro": "center",
    "meredithherald.com": "center",
    "dailytrust.com": "center",
    "mediafax.ro": "center",
    "assamtribune.com": "center",
    "paisano-online.com": "center",
    "variety.com": "center",
    "khybernews.tv": "center",
    "calvinchimes.org": "center",
    "ncregister.com": "right",
    "tn.com.ar": "center",
    "onlineindiannews.com": "center",
    "lexpress.fr": "lean left",
    "cherokeephoenix.org": "center",
    "blic.rs": "center",
    "manchestereveningnews.co.uk": "center",
    "elperiodico.cat": "lean left",
    "ilgiorno.it": "center",
    "dunya.com": "lean right",
    "chroniclelive.co.uk": "center",
    "dailysabah.com": "lean right",
    "dailyillini.com": "center",
    "ondacero.es": "center",
    "dailynigerian.com": "center",
    "channelstv.com": "center",
    "promptnewsonline.com": "center",
    "realnewsmagazine.net": "center",
    "dailysignal.com": "right",
    "eltiempo.com": "lean left",
    "expansion.mx": "center",
    "newtelegraphng.com": "center",
    "nos.nl": "center",
    "thepacer.net": "center",
    "inquisitr.com": "center",
    "antena3.com": "center",
    "thechronicle.com.gh": "center",
    "thedailybeast.com": "lean left",
    "ziuadevest.ro": "center",
    "buvoice.com": "center",
    "bzc.ro": "center",
    "sundayexpress.co.ls": "center",
    "radaronline.com": "center",
    "africachinapresscentre.org": "center",
    "handelszeitung.ch": "center",
    "southafricatoday.net": "lean right",
    "rnz.co.nz": "center",
    "aol.com": "center",
    "nzcity.co.nz": "center",
    "parkiet.com": "center",
    "berkshireeagle.com": "center",
    "n-tv.de": "center",
    "nigeriagalleria.com": "center",
    "brazzil.com": "center",
    "rutlandherald.com": "center",
    "thebatt.com": "center",
    "mlive.com": "center",
    "occidente.co": "center",
    "nigerianeye.com": "center",
    "kohajone.com": "center",
    "newsinvestigatorsng.com": "center",
    "diariopanorama.com": "center",
    "internazionale.it": "lean left",
    "sicilianews24.it": "center",
    "hola.com": "center",
    "freep.com": "lean left",
    "foxbusiness.com": "lean right",
    "georgeherald.com": "center",
    "lanacion.com.ar": "lean right",
    "newslaundry.com": "lean left",
    "aajtak.in": "center",
    "independentaustralia.net": "lean left",
    "dailytitan.com": "center",
    "euroweeklynews.com": "center",
    "dailystar.co.uk": "lean right",
    "elpueblodeceuta.es": "center",
    "guyanachronicle.com": "lean right",
    "namibian.com.na": "center",
    "lusakatimes.com": "center",
    "curierulnational.ro": "center",
    "romanialibera.ro": "lean right",
    "rg.ru": "right",
    "rhythm937.com": "center",
    "rnd.de": "center",
    "walesonline.co.uk": "center",
    "thaiexaminer.com": "center",
    "staradvertiser.com": "center",
    "diariodenavarra.es": "center",
    "bubblear.com": "center",
    "czechjournal.cz": "center",
    "grist.org": "lean left",
    "9news.com.au": "center",
    "punto.com.ph": "center",
    "csmonitor.com": "lean left",
    "levif.be": "center",
    "novinite.com": "center",
    "hurriyet.com.tr": "lean right",
    "chicagotribune.com": "lean right",
    "sabah.com.tr": "right",
    "adaderana.lk": "center",
    "globes.co.il": "center",
    "spiegel.de": "lean left",
    "manager-magazin.de": "center",
    "vetlandaposten.se": "center",
    "uiargonaut.com": "center",
    "theborneopost.com": "center",
    "rediff.com": "center",
    "scientificamerican.com": "lean left",
    "dispatch.com": "center",
    "pbs.org": "lean left",
    "nepalnews.com": "center",
    "mercado.com.ar": "center",
    "ovaciones.com": "center",
    "europapress.es": "center",
    "cope.es": "lean right",
    "valenciaplaza.com": "center",
    "stanforddaily.com": "center",
    "sobifm.com": "center",
    "elperiodicodeportivo.com.co": "center",
    "popmatters.com": "lean left",
    "lavozdeasturias.es": "center",
    "dailycal.org": "lean left",
    "standaard.be": "lean left",
    "ilmattino.it": "center",
    "ksl.com": "center",
    "wpolityce.pl": "right",
    "oe24.at": "lean right",
    "deutschlandfunk.de": "center",
    "charter97.org": "lean left",
    "tt.com": "center",
    "interia.pl": "center",
    "famagusta-gazette.com": "center",
    "panorama.it": "center",
    "emirates247.com": "center",
    "mxmindia.com": "center",
    "nationalenquirer.com": "right",
    "affarsvarlden.se": "center",
    "t-online.de": "center",
    "talkingpointsmemo.com": "lean left",
    "thecitizen.co.tz": "center",
    "politicsweb.co.za": "center",
    "thisisradionow.com": "center",
    "uzone.id": "center",
    "pagina12.com.ar": "left",
    "ieyenews.com": "center",
    "rosario3.com": "center",
    "egyptindependent.com": "center",
    "elciudadanoweb.com": "center",
    "nltimes.nl": "center",
    "lejdd.fr": "lean right",
    "inquirer.net": "lean left",
    "ew.com": "center",
    "nikkei.com": "center",
    "mainichi.jp": "lean left",
    "laopinioncoruna.es": "center",
    "prospect.org": "left",
    "thepostmillennial.com": "right",
    "thinkit.co.jp": "center",
    "ceskenoviny.cz": "center",
    "leparisien.fr": "center",
    "sport.es": "center",
    "laopiniondemalaga.es": "center",
    "lne.es": "center",
    "farodevigo.es": "center",
    "laopiniondemurcia.es": "center",
    "diariodeibiza.es": "center",
    "diariocordoba.com": "center",
    "elperiodicodearagon.com": "center",
    "laopiniondezamora.es": "center",
    "diariodemallorca.es": "center",
    "eldia.es": "center",
    "foothillscript.com": "center",
    "gazetadopovo.com.br": "right",
    "rtl.de": "center",
    "ajc.com": "lean left",
    "ilfattoquotidiano.it": "lean left",
    "bild.de": "lean right",
    "ft.com": "center",
    "rpereznetonline.com.ar": "center",
    "lindependant.fr": "center",
    "cambodgemag.com": "center",
    "kurir.rs": "lean right",
    "midilibre.fr": "center",
    "prensaeconomica.com.ar": "center",
    "mediaset.it": "center",
    "jurnalul.ro": "lean left",
    "express.co.uk": "lean right",
    "stirileprotv.ro": "center",
    "rtl.nl": "center",
    "politika.rs": "lean right",
    "nieuwsblad.be": "center",
    "bazonline.ch": "center",
    "derbund.ch": "center",
    "bernerzeitung.ch": "center",
    "tagesanzeiger.ch": "lean left",
    "okezone.com": "center",
    "thediplomat.ro": "center",
    "cartacapital.com.br": "left",
    "theweek.com": "center",
    "ripplesnigeria.com": "center",
    "rthk.hk": "center",
    "cnc3.co.tt": "center",
    "redstate.com": "right",
    "magic828.co.za": "center",
    "rp-online.de": "center",
    "beleggersbelangen.nl": "center",
    "sundiatapost.com": "center",
    "orientalnewsng.com": "center",
    "ilsecoloxix.it": "center",
    "larena.it": "center",
    "huelvainformacion.es": "center",
    "clubic.com": "center",
    "europasur.es": "center",
    "diariodecadiz.es": "center",
    "diariodejerez.es": "center",
    "granadahoy.com": "center",
    "malagahoy.es": "center",
    "diariodesevilla.es": "center",
    "eldiadecordoba.es": "center",
    "diariodealmeria.es": "center",
    "science-et-vie.com": "center",
    "phonemantra.com": "center",
    "ksusentinel.com": "center",
    "gazettenet.com": "center",
    "teletica.com": "center",
    "ebc.com.br": "center",
    "istoe.com.br": "lean left",
    "istoedinheiro.com.br": "center",
    "laprensalibre.cr": "center",
    "bota.al": "center",
    "cotidianul.ro": "lean right",
    "elconfidencial.com": "center",
    "lavoz.com.ar": "center",
    "ilgazzettino.it": "center",
    "startribune.com": "lean left",
    "sierraleonepress.com": "center",
    "castellonplaza.com": "center",
    "authorityngr.com": "center",
    "pointblanknews.com": "center",
    "theroot.com": "left",
    "thewrap.com": "center",
    "gmanetwork.com": "center",
    "thetrentonline.com": "center",
    "aftonbladet.se": "lean left",
    "hurriyetdailynews.com": "center",
    "buzzfeed.com": "lean left",
    "politicalwire.com": "lean left",
    "agi.it": "center",
    "larazon.es": "lean right",
    "dailyguidenetwork.com": "lean right",
    "filmfare.com": "center",
    "thegazette.news": "center",
    "infosanpedro.com.ar": "center",
    "limite42.com": "center",
    "santafederf.com.ar": "center",
    "fmcompromiso.com.ar": "center",
    "radioaoran.com.ar": "center",
    "reflejojujuy.com": "center",
    "elnuevodia.com": "lean left",
    "usmagazine.com": "center",
    "goo.ne.jp": "center",
    "axcessnews.com": "center",
    "ilgiornale.it": "lean right",
    "dubawa.org": "center",
    "worldtribune.com": "right",
    "alicanteplaza.es": "center",
    "tampabay.com": "lean left",
    "entornourgente.com": "center",
    "nationaldailyng.com": "center",
    "thenewcrusadingguideonline.com": "center",
    "qz.com": "lean left",
    "newstatesman.com": "left",
    "huntnewsnu.com": "center",
    "newsblaze.com": "right",
    "salon.com": "left",
    "eleconomista.com.mx": "center",
    "letelegramme.fr": "center",
    "kosmo.com.my": "center",
    "newsbomb.gr": "lean right",
    "wired868.com": "center",
    "clarin.com": "lean right",
    "diariovasco.com": "center",
    "stluciatimes.com": "center",
    "ideal.es": "center",
    "agora-web.jp": "center",
    "catalunyapress.cat": "center",
    "jis.gov.jm": "center",
    "mingpao.com": "center",
    "naijafeed.com": "center",
    "ukrinform.ua": "center",
    "newmail-ng.com": "center",
    "politicsnigeria.com": "center",
    "gistmania.com": "center",
    "augustachronicle.com": "lean right",
    "fakt.pl": "lean right",
    "polygon.com": "lean left",
    "sttinfo.fi": "center",
    "persecondnews.com": "center",
    "ozgurkocaeli.com.tr": "center",
    "irna.ir": "right",
    "eiga.com": "center",
    "antiguaobserver.com": "center",
    "indystar.com": "lean left",
    "recorder.com": "center",
    "renasterea.ro": "center",
    "ilna.ir": "lean left",
    "opinione.it": "lean right",
    "abstvradio.com": "center",
    "gov.ky": "center",
    "diariodeferrol.com": "center",
    "diariodearousa.com": "center",
    "economist.com": "center",
    "kauppalehti.fi": "center",
    "barbadosgazette.com": "center",
    "nationnews.com": "center",
    "moss-dagblad.no": "center",
    "cleveland.com": "center",
    "newsonjapan.com": "center",
    "gigazine.net": "center",
    "ig.com.br": "center",
    "tygodnikpowszechny.pl": "lean left",
    "newuniversity.org": "center",
    "insider.co.uk": "center",
    "ndtv.com": "lean left",
    "ara.cat": "lean left",
    "nnn.ng": "center",
    "varese7press.it": "center",
    "varesenews.it": "center",
    "metropoles.com": "center",
    "rionegro.com.ar": "center",
    "campusreview.com.au": "center",
    "pitchfork.com": "lean left",
    "kogiflame.com": "center",
    "lanuevacronica.com": "center",
    "lbccviking.com": "center",
    "dailyreviewonline-ng.com": "center",
    "ynetnews.com": "lean left",
    "salamanca24horas.com": "center",
    "stkittsgazette.com": "center",
    "wprost.pl": "lean right",
    "tagesschau.de": "center",
    "lagaceta.com.ar": "center",
    "oko.press": "lean left",
    "dailytimesng.com": "center",
    "eldebate.com.ar": "center",
    "eltribuno.com": "center",
    "adressa.no": "center",
    "tagblatt.ch": "center",
    "okay.ng": "center",
    "stvincenttribune.com": "center",
    "dominicagazette.com": "center",
    "guyanainquirer.com": "center",
    "grenadachronicle.com": "center",
    "dominicanrepublicpost.com": "center",
    "haitigazette.com": "center",
    "puertoricotribune.com": "center",
    "trinidadtribune.com": "center",
    "jamaicainquirer.com": "center",
    "bahamasspectator.com": "center",
    "stluciachronicle.com": "center",
    "superdeporte.es": "center",
    "trybuna.info": "left",
    "indipendenteonline.it": "center",
    "belami.rs": "center",
    "abril.com.br": "center",
    "vizionplus.tv": "center",
    "nowgrenada.com": "center",
    "dailykos.com": "left",
    "021.rs": "center",
    "rtv.rs": "center",
    "lafm.com.co": "center",
    "inews.co.uk": "lean left",
    "zuonline.ch": "center",
    "zsz.ch": "center",
    "landbote.ch": "center",
    "thunertagblatt.ch": "center",
    "weltexpress.info": "lean left",
    "kenyoncollegian.com": "center",
    "seattlespectator.com": "center",
    "pauljorion.com": "left",
    "brf.be": "center",
    "punchng.com": "center",
    "the-sun.com": "lean right",
    "lanazione.it": "center",
    "morningstaronline.co.uk": "left",
    "blikopnieuws.nl": "center",
    "saz-aktuell.com": "center",
    "pcccourier.com": "center",
    "jungewelt.de": "left",
    "20minutos.es": "center",
    "lapresse.ca": "center",
    "fordhamobserver.com": "center",
    "gettysburgian.com": "center",
    "dailydot.com": "lean left",
    "thestandard.com.hk": "center",
    "catalunyadiari.com": "center",
    "monitorulexpres.ro": "center",
    "tuoitrenews.vn": "center",
    "daylightng.com": "center",
    "politics.co.uk": "center",
    "sunstar.com.ph": "center",
    "pinkvilla.com": "center",
    "zdf.de": "center",
    "ilriformista.it": "lean left",
    "ticotimes.net": "center",
    "diariocritico.com": "center",
    "bernews.com": "center",
    "guernseypress.com": "center",
    "jerseyeveningpost.com": "center",
    "larepublica.co": "center",
    "wired.jp": "lean left",
    "tion.ro": "center",
    "thesun.co.uk": "lean right",
    "nrk.no": "center",
    "borneobulletin.com.bn": "center",
    "lastampa.it": "lean left",
    "polsatnews.pl": "lean right",
    "trend.az": "center",
    "elindependiente.com": "center",
    "eldia.com": "center",
    "technologyreview.jp": "center",
    "republika.co.id": "lean right",
    "neuquenalinstante.com.ar": "center",
    "surinenglish.com": "center",
    "csudhbulletin.com": "center",
    "liberoquotidiano.it": "lean right",
    "lagazzettadelmezzogiorno.it": "center",
    "gazeta.pl": "center",
    "lestimes.com": "center",
    "hallmarknews.com": "center",
    "people.cn": "right",
    "36ng.ng": "center",
    "valeursactuelles.com": "right",
    "ecodibergamo.it": "center",
    "lhrtimes.com": "center",
    "lalibre.be": "center",
    "thesimpsonian.com": "center",
    "thecrite.com": "center",
    "gwhatchet.com": "center",
    "hs.fi": "center",
    "rpdnoticias.com": "center",
    "aoc.media": "lean left",
    "sjuhawknews.com": "center",
    "nd.nl": "center",
    "uatrav.com": "center",
    "la-croix.com": "lean left",
    "socialsamosa.com": "center",
    "francetvinfo.fr": "center",
    "opencanada.org": "center",
    "reinformation.tv": "right",
    "newsbusters.org": "right",
    "berkeleybeacon.com": "center",
    "lanoticiadigital.com.ar": "center",
    "kpakpandofm.com.ng": "center",
    "tvcnews.tv": "center",
    "businessmirror.com.ph": "center",
    "numerama.com": "lean left",
    "naharnet.com": "center",
    "thestkittsnevisobserver.com": "center",
    "ladige.it": "center",
    "sueddeutsche.de": "lean left",
    "eldiario.es": "left",
    "dailystatesman.com.gh": "lean right",
    "aargauerzeitung.ch": "center",
    "elmercurio.com.mx": "center",
    "luzernerzeitung.ch": "center",
    "ariesonline.com.ar": "center",
    "maravipost.com": "center",
    "viva.co.id": "center",
    "lapatilla.com": "lean right",
    "jurnal.md": "center",
    "expressen.se": "lean right",
    "prlog.org": "center",
    "baylorlariat.com": "center",
    "hotair.com": "right",
    "vanderbilthustler.com": "center",
    "bellanaija.com": "center",
    "humanite.fr": "left",
    "ziarulnational.md": "center",
    "ghanaiantimes.com.gh": "center",
    "cugetliber.ro": "center",
    "jhunewsletter.com": "center",
    "byuiscroll.org": "center",
    "libertatea.ro": "lean left",
    "mainecampus.com": "center",
    "francesoir.fr": "lean right",
    "letemps.ch": "lean left",
    "ellitoral.com.ar": "center",
    "herault-tribune.com": "center",
    "thaipost.net": "lean right",
    "diariosur.es": "center",
    "suedostschweiz.ch": "center",
    "one.com.mt": "center",
    "kuenselonline.com": "center",
    "indiasnews.net": "center",
    "nigeriannewsdirect.com": "center",
    "davidsonian.com": "center",
    "netsecurity.ne.jp": "center",
    "que.es": "center",
    "ugandanews.net": "center",
    "theuknews.com": "center",
    "thetriangle.org": "center",
    "elespanol.com": "lean right",
    "britainnews.net": "center",
    "unitedkingdomnews.net": "center",
    "timescolonist.com": "center",
    "neusroom.com": "center",
    "asiatoday.co.kr": "center",
    "patriotpost.us": "right",
    "chinanews.net": "right",
    "ctvnews.ca": "center",
    "beijingnews.net": "right",
    "shanghainews.net": "right",
    "ibtimes.com.au": "center",
    "romaniajournal.ro": "center",
    "forbes.com.mx": "center",
    "thetech.com": "center",
    "lyoncapitale.fr": "center",
    "excite.co.jp": "center",
    "polityka.pl": "lean left",
    "kimberleybulletin.com": "center",
    "abbynews.com": "center",
    "trailtimes.ca": "center",
    "heraldo.es": "center",
    "amarujala.com": "center",
    "ign.com": "center",
    "dailymaverick.co.za": "lean left",
    "abcnyheter.se": "center",
    "nationalmemo.com": "lean left",
    "sn.se": "center",
    "aap.com.au": "center",
    "kkuriren.se": "center",
    "vietnamplus.vn": "center",
    "bizjournals.com": "center",
    "corriere.it": "center",
    "gazette.al": "center",
    "sciencesetavenir.fr": "center",
    "thelogicalindian.com": "lean left",
    "jamaicaobserver.com": "center",
    "freitag.de": "left",
    "business-standard.com": "center",
    "danas.rs": "lean left",
    "calgaryherald.com": "center",
    "news24.com": "center",
    "parade.com": "center",
    "rtlnieuws.nl": "center",
    "manoramaonline.com": "center",
    "citinewsroom.com": "center",
    "yenimesaj.com.tr": "lean right",
    "amsterdamnews.net": "center",
    "kbc.co.ke": "center",
    "cbc.bb": "center",
    "bizwatchnigeria.ng": "center",
    "thestar.com.my": "center",
    "google.com": "center",
    "haaretz.com": "lean left",
    "usnews.com": "center",
    "bssnews.net": "center",
    "gunosy.com": "center",
    "herald.ng": "center",
    "moneycontrol.com": "center",
    "wam.ae": "right",
    "freemalaysiatoday.com": "lean left",
    "france24.com": "center",
    "thehimalayantimes.com": "center",
    "themalaysianreserve.com": "center",
    "msn.com": "lean left",
    "phnompenhpost.com": "center",
    "newbusinessethiopia.com": "center",
    "pri.org": "lean left",
    "infobae.com": "center",
    "bq-magazine.com": "center",
    "marca.com": "center",
    "firstpost.com": "center",
    "nationaljournal.com": "center",
    "tribune-intl.com": "center",
    "businesschief.com": "center",
    "impress.co.jp": "center",
    "medium.com": "lean left",
    "forbesindia.com": "center",
    "tunein.com": "center",
    "laprovence.com": "center",
    "eluniversal.com.mx": "center",
    "monitor.co.ug": "center",
    "newsarawaktribune.com.my": "center",
    "theaggie.org": "center",
    "canadanews.net": "center",
    "gulf-times.com": "center",
    "gelocal.it": "center",
    "thewire.in": "lean left",
    "thesandspur.org": "center",
    "quintopoder.com.co": "center",
    "elcolombiano.com": "center",
    "chronicle.gi": "center",
    "limportant.fr": "center",
    "vov.vn": "center",
    "svt.se": "center",
    "pugetsound.edu": "center",
    "cadranpolitic.ro": "center",
    "canadastandard.com": "center",
    "scotsman.com": "center",
    "sudouest.fr": "center",
    "eurolibertes.com": "right",
    "singaporenews.net": "center",
    "tdg.ch": "center",
    "24heures.ch": "center",
    "estadao.com.br": "center",
    "bdnews24.com": "center",
    "bnr.bg": "center",
    "anglicanjournal.com": "center",
    "profil.at": "center",
    "rpi.edu": "center",
    "lest-eclair.fr": "center",
    "lunion.fr": "center",
    "thebutlercollegian.com": "center",
    "japantimes.co.jp": "center",
    "thepeninsulaqatar.com": "center",
    "armytimes.com": "center",
    "ghheadlines.com": "center",
    "airforcetimes.com": "center",
    "navytimes.com": "center",
    "statehornet.com": "center",
    "newzealandnews.net": "center",
    "indonesianews.net": "center",
    "jakartanews.net": "center",
    "reporteindigo.com": "center",
    "ourtimebd.com": "center",
    "wiwo.de": "center",
    "businessworld.in": "center",
    "uproxx.com": "lean left",
    "tiranatimes.com": "center",
    "groene.nl": "lean left",
    "nineoclock.ro": "center",
    "realitatea.net": "center",
    "aydinses.com": "center",
    "mndaily.com": "lean left",
    "australianherald.com": "center",
    "australiannews.net": "center",
    "lepoint.fr": "lean right",
    "rtl.be": "center",
    "thaipbs.or.th": "center",
    "brytfmonline.com": "center",
    "ojim.fr": "right",
    "hani.co.kr": "lean left",
    "fajar.co.id": "center",
    "courrierinternational.com": "lean left",
    "thesunflower.com": "center",
    "mbl.is": "center",
    "advancetitan.com": "center",
    "rttnews.com": "center",
    "unionesarda.it": "center",
    "thecostaricanews.com": "center",
    "tamilmurasu.com.sg": "center",
    "afaqs.com": "center",
    "paz.de": "center",
    "dailyeasternnews.com": "center",
    "dataphyte.com": "center",
    "haberturk.com": "lean right",
    "mybroadband.co.za": "center",
    "businesswire.com": "center",
    "laprovinciadicomo.it": "center",
    "theeastafrican.co.ke": "center",
    "philippinesnews.net": "center",
    "taipeinews.net": "center",
    "philippinetimes.com": "center",
    "am990formosa.com": "center",
    "mein-geld-medien.de": "center",
    "iol.co.za": "center",
    "opinionnigeria.com": "center",
    "people.com.cn": "right",
    "opednews.com": "lean left",
    "ottawacitizen.com": "center",
    "abidingtv.com": "center",
    "24-horas.mx": "center",
    "newsletter.co.uk": "center",
    "oradesibiu.ro": "center",
    "volksblatt.at": "center",
    "elclarinete.com.mx": "center",
    "thoughtleader.co.za": "lean left",
    "gazzetta.it": "center",
    "lfpress.com": "center",
    "rd.nl": "center",
    "theaimn.com": "lean left",
    "courrier.jp": "center",
    "flux.md": "center",
    "vlt.se": "center",
    "refinery29.com": "left",
    "huffingtonpost.co.uk": "lean left",
    "atlantico.fr": "lean right",
    "monitorfg.ro": "center",
    "unzensuriert.at": "right",
    "boerse-online.de": "center",
    "salzburg24.at": "center",
    "tribuneindia.com": "center",
    "oneindia.com": "center",
    "ilfriuli.it": "center",
    "italpress.com": "center",
    "petra.gov.jo": "right",
    "mondo.rs": "center",
    "focustaiwan.tw": "center",
    "shn.ch": "center",
    "birminghammail.co.uk": "center",
    "ismedia.jp": "center",
    "samoanews.com": "center",
    "canarias7.es": "center",
    "standard.rs": "center",
    "opensecrets.org": "center",
    "telemadrid.es": "center",
    "trud.ru": "lean right",
    "dispatchlive.co.za": "center",
    "pmnewsnigeria.com": "center",
    "argentinastar.com": "center",
    "buenosairesnews.net": "center",
    "argentinanews.net": "center",
    "ncronline.org": "lean left",
    "ildispariquotidiano.it": "center",
    "americanfreepress.net": "right",
    "khaosod.co.th": "center",
    "modernghana.com": "center",
    "thelocal.se": "center",
    "cdns.com.tw": "center",
    "commentcamarche.net": "center",
    "ambito.com": "center",
    "goldengatexpress.org": "center",
    "sagat.no": "center",
    "techable.jp": "center",
    "limburger.nl": "center",
    "lagazettedescommunes.com": "center",
    "elbistaninsesi.com": "center",
    "cumhuriyet.com.tr": "lean left",
    "broadview.org": "lean left",
    "diariodeleon.es": "center",
    "lesoir.be": "lean left",
    "publico.es": "lean left",
    "reddeeradvocate.com": "center",
    "jomhornews.com": "center",
    "graphic.com.gh": "center",
    "macleans.ca": "center",
    "dohanews.co": "center",
    "aktiv-online.de": "center",
    "dailymirror.lk": "center",
    "3-mob.com": "center",
    "efe.com": "center",
    "wallstreetitalia.com": "center",
    "dagbladet.no": "lean left",
    "20minutes.fr": "center",
    "troymedia.com": "center",
    "handelsblatt.com": "center",
    "elconfidencialdigital.com": "center",
    "thenewdaily.com.au": "lean left",
    "smp.no": "center",
    "ictnews.org": "center",
    "lavozdigital.es": "center",
    "boersen-zeitung.de": "center",
    "elcorreogallego.es": "center",
    "thephuketnews.com": "center",
    "perfil.com": "center",
    "cshoy24.com.ar": "center",
    "pattayamail.com": "center",
    "hbl.fi": "center",
    "aravot-en.am": "center",
    "ekuriren.se": "center",
    "moneyweb.co.za": "center",
    "noticierouniversal.com": "center",
    "nu.nl": "center",
    "wartaekonomi.co.id": "center",
    "thedepauw.com": "center",
    "loyolamaroon.com": "center",
    "ynaija.com": "lean left",
    "parool.nl": "center",
    "dailytarheel.com": "lean left",
    "liberation.fr": "left",
    "cbn.co.za": "center",
    "jungle.world": "left",
    "vanguardngr.com": "center",
    "salamancartvaldia.es": "center",
    "kommunal-rapport.no": "center",
    "pulse.ng": "lean left",
    "dailynexus.com": "lean left",
    "ariananews.af": "center",
    "chargerbulletin.com": "center",
    "iomtoday.co.im": "center",
    "kommersant.ru": "lean right",
    "observadorcentral.com.ar": "center",
    "rnanews.com": "center",
    "elimparcial.com": "center",
    "rfi.fr": "center",
    "worldtimes.co.jp": "center",
    "aristeguinoticias.com": "lean left",
    "alba24.ro": "center",
    "tijd.be": "center",
    "sydsvenskan.se": "lean left",
    "radiogdansk.pl": "center",
    "linternaute.com": "center",
    "hd.se": "center",
    "zdnet.fr": "center",
    "rtr.ch": "center",
    "lexo.al": "center",
    "nouvelobs.com": "lean left",
    "hln.be": "center",
    "newsclick.in": "lean left",
    "destentor.nl": "center",
    "bndestem.nl": "center",
    "pzc.nl": "center",
    "ad.nl": "center",
    "ed.nl": "center",
    "bd.nl": "center",
    "tubantia.nl": "center",
    "gelderlander.nl": "center",
    "ndsmcobserver.com": "center",
    "dbknews.com": "center",
    "oregonlive.com": "lean left",
    "capital.fr": "center",
    "indiewire.com": "lean left",
    "nacion.com": "center",
    "milenio.com": "center",
    "radionigeria.gov.ng": "center",
    "tuftsdaily.com": "center",
    "bienpublic.com": "center",
    "lalsace.fr": "center",
    "7sur7.be": "center",
    "tf1info.fr": "center",
    "dn.no": "center",
    "nextinpact.com": "lean left",
    "opindia.com": "right",
    "vozpopuli.com": "center",
    "thepanamanews.com": "lean left",
    "abante.com.ph": "center",
    "remate.ph": "center",
    "eurweb.com": "lean left",
    "herald.co.zw": "lean right",
    "newsonair.com": "center",
    "noticiasvigo.es": "center",
    "barcelonanews.net": "center",
    "thesundaily.my": "center",
    "dailyuw.com": "center",
    "fijitimes.com": "center",
    "nationalnewswatch.com": "lean left",
    "brusselstimes.com": "center",
    "demorgen.be": "lean left",
    "ledauphine.com": "center",
    "lejsl.com": "center",
    "leprogres.fr": "center",
    "republicain-lorrain.fr": "center",
    "vosgesmatin.fr": "center",
    "estrepublicain.fr": "center",
    "kenyan-post.com": "lean right",
    "misionesonline.net": "center",
    "dziennikbaltycki.pl": "center",
    "naszemiasto.pl": "center",
    "elsoldemexico.com.mx": "center",
    "elbravo.mx": "center",
    "alahednews.com.lb": "right",
    "radio.opole.pl": "center",
    "gazetakrakowska.pl": "center",
    "kurierlubelski.pl": "center",
    "dzienniklodzki.pl": "center",
    "expressilustrowany.pl": "center",
    "nowiny24.pl": "center",
    "nowosci.com.pl": "center",
    "dziennikzachodni.pl": "center",
    "gazetalubuska.pl": "center",
    "pomorska.pl": "center",
    "gloswielkopolski.pl": "center",
    "wspolczesna.pl": "center",
    "poranny.pl": "center",
    "dziennikpolski24.pl": "center",
    "prachachat.net": "center",
    "usforacle.com": "center",
    "e-vid.ru": "right",
    "spectatornews.com": "center",
    "tv2.no": "center",
    "theabujainquirer.com": "center",
    "echodnia.eu": "center",
    "helsinkitimes.fi": "center",
    "ewn.co.za": "center",
    "volkskrant.nl": "lean left",
    "allanyheter.com": "center",
    "scrippsnews.com": "center",
    "eikernytt.no": "center",
    "antiwar.com": "lean left",
    "thelocal.no": "center",
    "folha.com.br": "lean left",
    "leblogalupus.com": "right",
    "trouw.nl": "lean left",
    "latin-america.jp": "center",
    "lainformacion.com": "center",
    "plenglish.com": "lean left",
    "woz.ch": "center",
    "bnr.nl": "center",
    "stripes.com": "center",
    "lesechos.fr": "center",
    "rmol.co": "center",
    "americanindependent.com": "lean left",
    "sfgate.com": "lean left",
    "gva.be": "center",
    "hbvl.be": "center",
    "thefridaytimes.com": "lean left",
    "abclive.in": "center",
    "iotnews.jp": "center",
    "vocedimantova.it": "center",
    "tvazteca.com": "center",
    "leparisien.fr": "center",
    "heraldlive.co.za": "center",
    "theweek.co.uk": "center",
    "bestnaira.com": "center",
    "tempo.co": "center",
    "eyeradio.org": "center",
    "nbr.co.nz": "center",
    "pakistannews.net": "center",
    "dhnet.be": "center",
    "ilgiornaledivicenza.it": "center",
    "bresciaoggi.it": "center",
    "huffingtonpost.fr": "lean left",
    "nieuws.nl": "center",
    "ledevoir.com": "lean left",
    "notretemps.com": "center",
    "interfax.ru": "right",
    "rsi.ch": "center",
    "top-channel.tv": "center",
    "metronews.it": "center",
    "omni.se": "center",
    "radio-canada.ca": "lean left",
    "publimetro.com.mx": "center",
    "elimparcial.es": "center",
    "journaldemontreal.com": "lean right",
    "liderweb.mx": "center",
    "sinembargo.mx": "lean left",
    "closeup.mx": "lean left",
    "ucol.mx": "center",
    "diario.mx": "center",
    "sipse.com": "center",
    "enfoqueinformativo.mx": "center",
    "nigerianobservernews.com": "center",
    "eldiariodechihuahua.mx": "center",
    "lavozdemichoacan.com.mx": "center",
    "urgente24.com": "lean right",
    "elwaha-dz.com": "center",
    "journaldequebec.com": "lean right",
    "proceso.com.mx": "lean left",
    "inform.kz": "center",
    "tempi.it": "right",
    "geo.fr": "center",
    "diario16.com": "lean left",
    "diariosanrafael.com.ar": "center",
    "capetalk.co.za": "center",
    "topalbaniaradio.com": "center",
    "vanguardia.com.mx": "center",
    "sopitas.com": "lean left",
    "iltalehti.fi": "center",
    "chinanews.com": "right",
    "sie7edechiapas.com": "center",
    "plazadearmas.com.mx": "center",
    "radiokontakt.al": "center",
    "livedoor.com": "center",
    "lebanon24.com": "center",
    "bisnis.com": "center",
    "dna.fr": "center",
    "mundopoliticodiario.com": "center",
    "srad.jp": "center",
    "jawapos.com": "center",
    "brasil247.com": "lean left",
    "taiwandaily.net": "center",
    "cnews.fr": "lean right",
    "nna-leb.gov.lb": "right",
    "channel4.com": "lean left",
    "falkopingstidning.se": "center",
    "skaraborgslanstidning.se": "center",
    "skovdenyheter.se": "center",
    "vimmerbytidning.se": "center",
    "vt.se": "center",
    "nwt.se": "center",
    "gd.se": "center",
    "sverigesradio.se": "center",
    "yohaig.ng": "center",
    "arbetarbladet.se": "lean left",
    "l1.nl": "center",
    "voiceofnaija.ng": "center",
    "wienerzeitung.at": "center",
    "tv4.se": "center",
    "thelocal.fr": "center",
    "europe1.fr": "lean right",
    "sudinfo.be": "center",
    "francebleu.fr": "center",
    "hket.com": "center",
    "yam.md": "center",
    "cicero.de": "lean right",
    "moldova.org": "center",
    "barrons.com": "center",
    "cameroononline.org": "center",
    "themonthly.com.au": "lean left",
    "dvhn.nl": "center",
    "forbesjapan.com": "center",
    "beobachter.ch": "center",
    "giornaledibrescia.it": "center",
    "weeklycuttingedge.com": "lean right",
    "elmundofinanciero.com": "center",
    "newsday.com.ng": "center",
    "meridiano.mx": "center",
    "vertigopolitico.com": "center",
    "seychellesnewsagency.com": "center",
    "portafolio.co": "center",
    "konbini.com": "lean left",
    "eldiario.com.co": "center",
    "jornada.com.mx": "lean left",
    "eleconomista.com.ar": "center",
    "liputan6.com": "center",
    "elintransigente.com": "center",
    "yourgibraltartv.com": "center",
    "imparcialoaxaca.mx": "center",
    "varldenidag.se": "lean right",
    "askanews.it": "center",
    "chacoprensa.com": "center",
    "slate.fr": "lean left",
    "indiantelevision.com": "center",
    "ilrestodelcarlino.it": "center",
    "slobodnaevropa.org": "lean left",
    "rabble.ca": "left",
    "ladepeche.fr": "center",
    "astaga.com": "center",
    "glasamerike.net": "center",
    "kuwaitbulletin.com": "center",
    "khmertimeskh.com": "center",
    "kentwired.com": "center",
    "nyheter24.se": "lean left",
    "bignewsnetwork.com": "center",
    "sunnewsonline.com": "lean right",
    "kaieteurnewsonline.com": "center",
    "mwnation.com": "center",
    "bt.se": "center",
    "nexojornal.com.br": "lean left",
    "aftenposten.no": "center",
    "thehilltoponline.com": "center",
    "guadanews.es": "center",
    "depauliaonline.com": "center",
    "rebelnews.ie": "right",
    "telam.com.ar": "lean left",
    "thedailyguardian.net": "center",
    "newswirengr.com": "center",
    "themiamihurricane.com": "center",
    "postcourier.com.pg": "center",
    "nationalaccordnewspaper.com": "center",
    "catamarcaprovincia.com.ar": "center",
    "laverdad.es": "center",
    "elnortedecastilla.es": "center",
    "lasprovincias.es": "center",
    "elcomercio.es": "center",
    "leonoticias.com": "center",
    "hoy.es": "center",
    "eldiariomontanes.es": "center",
    "larioja.com": "center",
    "telerama.fr": "center",
    "maraviexpress.com": "center",
    "burgosconecta.es": "center",
    "nyasatimes.com": "lean left",
    "gazeta.ru": "lean right",
    "avantionline.it": "center",
    "extra.com.co": "center",
    "diariodelcauca.com.co": "center",
    "elpais.cr": "center",
    "arcturius.org": "lean left",
    "etownian.com": "center",
    "thelocal.de": "center",
    "kualalumpurnews.net": "center",
    "nigeriasun.com": "center",
    "cambodiantimes.com": "center",
    "afghanistansun.com": "center",
    "novosti.rs": "lean right",
    "eriepa.com": "center",
    "canadianbusiness.com": "center",
    "suara.com": "center",
    "nicematin.com": "center",
    "varmatin.com": "center",
    "hellomagazine.com": "center",
    "thecarletonian.com": "center",
    "volanteonline.com": "center",
    "digiday.jp": "center",
    "nifty.com": "center",
    "thairath.co.th": "center",
    "cndigital.com.ar": "center",
    "wyborcza.pl": "lean left",
    "laloyolan.com": "center",
    "lobosnews.com.ar": "center",
    "lapoliticaonline.com": "lean left",
    "aip.nu": "center",
    "gnnliberia.com": "center",
    "24matins.fr": "center",
    "milliyet.com.tr": "lean right",
    "thecambodianews.net": "center",
    "naslovi.net": "center",                 
    "elnacional.cat": "lean left"            
}

In [5]:
#extract domain from the 'media_url' column
df["domain"] = df["media_url"].str.extract(r"(?:https?:\/\/)?(?:www\.)?([^\/]+)")

In [6]:
#map each domain in the dataset to its political orientation using the bias_map list
#if a domain is not found in bias_map, it will be "unknown"
def bias_lookup(domain):
    return bias_map.get(domain, "unknown")

df["political_orientation"] = df["domain"].map(bias_lookup)

### Results: Political Orientation Distribution

The following section shows the counts of each orientation in our dataset.  
This distribution highlights whether the dataset is balanced or skewed toward certain perspectives.

In [7]:
#the distribution of political orientation
print("Political Orientation Distribution:")
print(df["political_orientation"].value_counts())

#ehe total number of outlets analyzed
print("\nTotal outlets analyzed:", len(df))

Political Orientation Distribution:
center        23271
lean left      8214
lean right     1515
right          1285
left           1114
Name: political_orientation, dtype: int64

Total outlets analyzed: 35399


In [8]:
#list of domains in the dataset that are not present in the bias_map
missing_domains = df.loc[~df["domain"].isin(bias_map.keys()), "domain"].unique()
print("Missing domains:", missing_domains[:10])

Missing domains: []


For domains that is not listed in these reference sites, we used **Gemini/GenAI** to query the missing domain directly. The model provided the political orientation (center, lean-left, left, lean-right, right) so that every domain in the dataset could be classified consistently.

In [9]:
from google import genai

client = genai.Client()

response = client.models.generate_content(
    model="gemini-2.5-flash", contents="what is the political orientation (central, lean left, left, lean right, right) answer only, of these domains: thecambodianews.net, naslovi.net, elnacional.cat"
)
print(response.text)

thecambodianews.net: central
naslovi.net: central
elnacional.cat: lean left


And then from the GenAI results, we updated the **bias_map** lists until there's no 'unknown' domains left.

### **Result Summary:** 
The dataset is dominated by outlets classified as **center (23,271 entries)**, with a much smaller share **lean-left (8,214)**, and very few **left (1,114)** or **right (1,285)**. **Lean-right** is also relatively **small (1,515)**. This means your collection is heavily weighted toward **centrist or neutral** sources.

---

### Interpretation
- **Balance:** The dataset is not evenly distributed. About **70%+ are “center”**, which may reflect the fact that many regional, business, or wire-service outlets are coded as neutral.
- **Asymmetry:** There are **more lean-left outlets than lean-right** (8,214 vs. 1,515). This could be due to the inclusion of many European and Latin American sources, where progressive editorial lines are more common.
- **Extremes are rare:** Both “left” and “right” categories are small, showing that strongly partisan outlets are underrepresented compared to centrist or mildly leaning ones.

---

### Reference
Media bias classification is often based on projects like **[AllSides Media Bias Ratings](https://www.allsides.com/media-bias/media-bias-ratings)**, **[Ad Fontes Media](https://app.adfontesmedia.com/chart/interactive)**, and **[Ground News](https://ground.news/bias-bar)**, which categorize outlets into *left, lean-left, center, lean-right, right* based on editorial stance, language, and ownership.  
For example, AllSides explains that:
- *Center* outlets “do not predictably favor either side of the political spectrum.”  
- *Lean-left* or *lean-right* outlets show consistent but mild bias in story selection or framing.  
- *Left* and *Right* outlets show strong ideological framing and selective coverage.  
